pandas/_libs/testing.pyx

```diff
@@ -24,7 +24,13 @@ from pandas.core.dtypes.missing import array_equivalent
 
 
 cdef bint isiterable(obj):
-    return hasattr(obj, "__iter__")
+    if hasattr(obj, "__iter__"):
```

Comment:
@jbrockmendel ok here?

Reply:
It doesn't seem to affect the runtime of the test suite by much